app/models/project_association.rb

```diff
@@ -37,7 +37,8 @@ class ProjectAssociation < ActiveRecord::Base
   belongs_to :project_b, class_name:  'Project',
                          foreign_key: 'project_b_id'
 
-  validates_presence_of :project_a, :project_b
+  alias_attribute :dependent_project, :project_b
+  validates_presence_of :project_a, :dependent_project
```

Comment:
1. probably @houndci knows things that we should take into account
2. If I understand this correctly you are aliasing `project_b` into `dependent` project to be able to give it a nice translation. Why don't you  do the same for `project_a`?